sdk/servicebus/service-bus/CHANGELOG.md

```diff
@@ -2,6 +2,9 @@
 
 ## 2.0.0-preview.1 (Unreleased)
 
+- Fixes [bug 7598][https://github.com/azure/azure-sdk-for-js/issues/7958] where the dead letter error and description could be populated
```

Comment:
Its still not clear to me how the changes in this PR fixes the bug 7598
Can you elaborate the root cause for this bug?

Reply:
Look at this spot here:

https://github.com/Azure/azure-sdk-for-js/blob/399aa7320fb7db51ad7cb03c61d67a32bdfd6503/sdk/servicebus/service-bus/src/serviceBusMessage.ts#L1023

Reply:
Ah! Because `condition` and `description` on the error object would be undefined. Got it.